internal/app/connectconformance/server_runner.go

```diff
@@ -26,8 +26,145 @@ import (
 	"connectrpc.com/conformance/internal"
 	conformancev1alpha1 "connectrpc.com/conformance/internal/gen/proto/go/connectrpc/conformance/v1alpha1"
 	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
 )
 
+// populateExpectedRequests populates the requests we expected to get back from the server
+// by examining the requests we sent.
+func populateExpectedRequests(testCase *conformancev1alpha1.TestCase) error {
+	// If an expected response was already provided, return and use that.
+	// This allows for overriding this function with explicit values in the yaml file.
+	if testCase.ExpectedResponse != nil {
+		return nil
+	}
+	if len(testCase.Request.RequestMessages) == 0 {
+		return errors.New("at least one request is required")
+	}
```

Comment:
I don't think we want this constraint. We want to be able to test client and bidi streams where there are no upload messages.

Maybe add a TODO and then we can follow-up with updates to the server impls to produce (and the code below to expect) a single response message in this situation, where the response data value is some fixed string (such as `"no response definition"`) and whose request info will still be present, but we expect it to indicate zero request messages.

In the future, we could come up with a very simple scheme for how the server can read a minimal response definition from request headers.

(Just riffing now...)

Instead of a full response definition, it could be something very simple like:
```
x-response-definition: foo:bar, fu:baz, 3:abcdef, 9:some error message
```
Where the value is a simple comma-delimited string with four components:
1. Pipe-delimited custom headers. Each header is in `name:value` form. So `foo:bar|foo:baz|frob:nitz` is equivalent to
   ```yaml
   [
      {name: "foo", value: ["bar", "baz"]},
      {name: "frob", value:["nitz"]}
   ]
   ```
   To keep parsing simple, the header values may not contain pipes or commas.
2. Pipe-delimited custom trailers.
3. Response messages in the form `count:data`. The count must be zero or one for client-stream endpoints, depending on if next field indicates an error or not. So `3:foo` will send back three response messages where the `data` field is `"foo"` (as bytes). The data value may not contain commas (to keep parsing of the definition simple).
4. Error in the form `code:message`. This string will be the simple value `"ok"` to mean no error. The code portion will be the numeric value of the gRPC code to use.

Reply:
I left a few TODOs in the code here to address this. Both places may be redundant, but at least we'll know where to make sure to cover our bases.